### PR TITLE
Add greedy_match() and greedy_substring_match()

### PR DESCRIPTION
**Pull request by ShaneSmit**
`greedy_match()` already existed within the cpp code, but `greedy_substring_match()` is new.  Both return multiple results.  However, `greedy_match()` uses prefix matching, while `greedy_substring_match()` uses substring matching.

### CODE REVIEW

**Comment by abrandoned** [comment]
@ShaneSmit great, can you provide examples of greedy/non-greedy matches in the Readme; I need to update the Readme in general as well so we have understanding of what the lib does

File: lib/ffi/radix_tree.rb [hunk 1]
```diff
@@ -135,6 +139,20 @@ def get(key)
         ::FFI::RadixTree.match_free(byte_pointer) if byte_pointer
       end
 
+      def set(key, value)
```
Comment (abrandoned) [inline lib/ffi/radix_tree.rb]: we already have `push` and now `set`; I think we should name it `update` or `swap` or `replace` ... just something that communicates the action better

**Comment by abrandoned** [comment]
@ShaneSmit other than naming I'm good

File: lib/ffi/radix_tree.rb (hunk 1 above)
Comment (ShaneSmit) [reply lib/ffi/radix_tree.rb]: I was trying to make a `set` to go along with the `get`.   `get` only exists on the Ruby side, it calls `fetch` on the c++ side.  So my `set` function in Ruby calls `update` in c++.  But I can change them to both be `update` if you'd like.

**Comment by ShaneSmit** [comment]
@abrandoned I renamed `set` to `push_or_update` as per our conversation.

**Pull request merged**